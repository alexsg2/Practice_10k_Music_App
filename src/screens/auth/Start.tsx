import React from 'react';
import { useNavigation } from '@react-navigation/native';
import { View, Image, StyleSheet, TouchableOpacity, Text } from 'react-native';

const StartScreen = () =>
{
    const navigation = useNavigation();
    
    return (
        <View style={styles.container}>
            <View style={styles.logoContainer}>
                <Image source={require('../../assets/images/white-logo.png')} style={styles.logo}/>
                <Text style={styles.headerText}>All-in-One Practice Hub.</Text>
            </View>
            <View style={styles.buttonContainer}>
                <TouchableOpacity onPress={() => navigation.navigate('Login')} style={styles.button}>
                    <Text style={styles.buttonText}>Login</Text>
                </TouchableOpacity>
                <TouchableOpacity onPress={() => navigation.navigate('Register')} style={styles.button}>
                    <Text style={styles.buttonText}>Register</Text>
                </TouchableOpacity>
            </View>
        </View>
    );
}

export default StartScreen;

const styles = StyleSheet.create(
{
    container: {
        flex: 1,
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: '#5982C2',
    },
    logoContainer: {
        width: '80%',
        alignItems: 'center',
        justifyContent: 'center',
    },
    logo: {
        aspectRatio: 1,
        marginTop: '30%',
    },
    headerText: {
        fontSize: 30,
        color: 'white',
        textAlign: 'center',
        marginVertical: '10%',
        // TODO: get a cursive font style
    },
    buttonContainer: {
        width: '80%',
        marginBottom: '20%',
        alignItems: 'center',
        justifyContent: 'center',
    },
    button: {
        width: '100%',
        padding: '7%',
        borderRadius: 10,
        marginVertical: '5%',
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: 'rgba(0, 0, 0, 0.7)',
    },
    buttonText: {
        fontSize: 18,
        color: 'white',
        fontWeight: 'bold',
    },
});
