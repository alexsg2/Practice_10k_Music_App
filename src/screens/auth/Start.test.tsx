import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { useNavigation, NavigationContainer } from '@react-navigation/native';
import Start from './Start';

// Mock the useNavigation hook
jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: jest.fn(),
}));

describe('<Start />', () => {
  it('renders correctly', () => {
    const tree = render(
      <NavigationContainer>
        <Start />
      </NavigationContainer>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('navigates to Login when the Login button is pressed', () => {
    // Setup the mock
    const mockNavigate = jest.fn();
    (useNavigation as jest.Mock).mockReturnValue({ navigate: mockNavigate });

    const { getByText } = render(
      <NavigationContainer>
        <Start />
      </NavigationContainer>
    );

    // Click login button
    fireEvent.press(getByText('Login'));

    expect(mockNavigate).toHaveBeenCalledWith('Login');
  });

  it('navigates to Register when the Register button is pressed', () => {
    // Setup the mock
    const mockNavigate = jest.fn();
    (useNavigation as jest.Mock).mockReturnValue({ navigate: mockNavigate });

    const { getByText } = render(
      <NavigationContainer>
        <Start />
      </NavigationContainer>
    );

    // Click register button
    fireEvent.press(getByText('Register'));

    expect(mockNavigate).toHaveBeenCalledWith('Register');
  });
});
