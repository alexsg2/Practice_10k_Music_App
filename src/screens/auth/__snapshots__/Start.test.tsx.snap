// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Start /> renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#5982C2",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
        "width": "80%",
      }
    }
  >
    <Image
      source={1}
      style={
        {
          "aspectRatio": 1,
          "marginTop": "30%",
        }
      }
    />
    <Text
      style={
        {
          "color": "white",
          "fontSize": 30,
          "marginVertical": "10%",
          "textAlign": "center",
        }
      }
    >
      All-in-One Practice Hub.
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
        "marginBottom": "20%",
        "width": "80%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.7)",
          "borderRadius": 10,
          "justifyContent": "center",
          "marginVertical": "5%",
          "opacity": 1,
          "padding": "7%",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 18,
            "fontWeight": "bold",
          }
        }
      >
        Login
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.7)",
          "borderRadius": 10,
          "justifyContent": "center",
          "marginVertical": "5%",
          "opacity": 1,
          "padding": "7%",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 18,
            "fontWeight": "bold",
          }
        }
      >
        Register
      </Text>
    </View>
  </View>
</View>
`;
